File: samples/python_equiv/data_heavy.py
class Unit:
    def __init__(self, name, hp, atk, defense):
        self.name = name
        self.hp = hp
        self.atk = atk
        self.defense = defense

class Skill:
    def __init__(self, dmg, cost):
        self.dmg = dmg
        self.cost = cost

heroes = [
    Unit("Knight", 120, 30, 20),
    Unit("Mage", 80, 50, 10),
    Unit("Rogue", 90, 40, 15),
]

skills = {
    "slash": Skill(25, 10),
    "fireball": Skill(50, 30),
    "backstab": Skill(35, 15),
}

def total_hp(units):
    s = 0
    for u in units:
        s += u.hp
    return s

def strongest(units):
    best = units[0]
    for u in units:
        if u.atk > best.atk:
            best = u
    return best

print("Total HP:", total_hp(heroes))
print("Strongest:", strongest(heroes).name)

File: samples/csharp_equiv/data_heavy.cs
using System;
using System.Collections.Generic;
using System.Linq;

class Unit
{
    public string Name { get; }
    public int Hp { get; }
    public int Atk { get; }
    public int Def { get; }

    public Unit(string name, int hp, int atk, int def)
    {
        Name = name;
        Hp = hp;
        Atk = atk;
        Def = def;
    }
}

class Skill
{
    public int Dmg { get; }
    public int Cost { get; }

    public Skill(int dmg, int cost)
    {
        Dmg = dmg;
        Cost = cost;
    }
}

class Program
{
    static int TotalHp(List<Unit> units) => units.Sum(u => u.Hp);

    static Unit Strongest(List<Unit> units) =>
        units.OrderByDescending(u => u.Atk).First();

    static void Main()
    {
        var heroes = new List<Unit>
        {
            new Unit("Knight", 120, 30, 20),
            new Unit("Mage", 80, 50, 10),
            new Unit("Rogue", 90, 40, 15),
        };

        var skills = new Dictionary<string, Skill>
        {
            ["slash"] = new Skill(25, 10),
            ["fireball"] = new Skill(50, 30),
            ["backstab"] = new Skill(35, 15),
        };

        Console.WriteLine($"Total HP: {TotalHp(heroes)}");
        Console.WriteLine($"Strongest: {Strongest(heroes).Name}");
    }
}
